<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="en">
<head>
  <!-- hexo-inject:begin --><!-- hexo-inject:end --><meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />







<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.4" rel="stylesheet" type="text/css" />


  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=5.1.4">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=5.1.4">


  <link rel="mask-icon" href="/images/logo.svg?v=5.1.4" color="#222">





  <meta name="keywords" content="Dry_lab,RNA-seq,Aligner," />










<meta name="description" content="Owing to the stay-at-home order, I cannot do my wet lab experiment anymore. To use this time effectively, I revisited my Mimulus time course data and thought through the questions I listed in the to-d">
<meta property="og:type" content="article">
<meta property="og:title" content="Finding suitable aligners for my project">
<meta property="og:url" content="https://loubya.github.io/2020/07/02/20200702-Finding-suitable-aligners-for-my-project/index.html">
<meta property="og:site_name" content="Po-Kai&#39;s Note">
<meta property="og:description" content="Owing to the stay-at-home order, I cannot do my wet lab experiment anymore. To use this time effectively, I revisited my Mimulus time course data and thought through the questions I listed in the to-d">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2020-07-02T07:06:00.000Z">
<meta property="article:modified_time" content="2020-07-02T07:36:46.769Z">
<meta property="article:author" content="Po-Kai">
<meta property="article:tag" content="Dry_lab">
<meta property="article:tag" content="RNA-seq">
<meta property="article:tag" content="Aligner">
<meta name="twitter:card" content="summary">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    version: '5.1.4',
    sidebar: {"position":"right","display":"post","offset":12,"b2t":true,"scrollpercent":true,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    duoshuo: {
      userId: '0',
      author: 'Author'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="https://loubya.github.io/2020/07/02/20200702-Finding-suitable-aligners-for-my-project/"/>





  <title>Finding suitable aligners for my project | Po-Kai's Note</title>
  








<meta name="generator" content="Hexo 4.2.0"><!-- hexo-inject:begin --><!-- hexo-inject:end --></head>

<body itemscope itemtype="http://schema.org/WebPage" lang="en">

  
  
    
  

  <!-- hexo-inject:begin --><!-- hexo-inject:end --><div class="container sidebar-position-right page-post-detail">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">Po-Kai's Note</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle">A Ph.D. student's random writing</p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            Home
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            About
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            Tags
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            Categories
          </a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-archive"></i> <br />
            
            Archives
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="https://loubya.github.io/2020/07/02/20200702-Finding-suitable-aligners-for-my-project/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Po-Kai">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="https://i.imgur.com/gJGAo55.jpg">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Po-Kai's Note">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">Finding suitable aligners for my project</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">Posted on</span>
              
              <time title="Post created" itemprop="dateCreated datePublished" datetime="2020-07-02T00:06:00-07:00">
                2020-07-02
              </time>
            

            

            
          </span>

          
            <span class="post-category" >
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">In</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/RNA-seq/" itemprop="url" rel="index">
                    <span itemprop="name">RNA-seq</span>
                  </a>
                </span>

                
                
                  , 
                
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/RNA-seq/Dry-lab/" itemprop="url" rel="index">
                    <span itemprop="name">Dry_lab</span>
                  </a>
                </span>

                
                
              
            </span>
          

          
            
              <span class="post-comments-count">
                <span class="post-meta-divider">|</span>
                <span class="post-meta-item-icon">
                  <i class="fa fa-comment-o"></i>
                </span>
                <a href="/2020/07/02/20200702-Finding-suitable-aligners-for-my-project/#comments" itemprop="discussionUrl">
                  <span class="post-comments-count disqus-comment-count"
                        data-disqus-identifier="2020/07/02/20200702-Finding-suitable-aligners-for-my-project/" itemprop="commentCount"></span>
                </a>
              </span>
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>Owing to the stay-at-home order, I cannot do my wet lab experiment anymore. To use this time effectively, I revisited my Mimulus time course data and thought through the questions I listed in the to-do note. I found that I still don’t have good answers for most of the questions, even though those are frequently brought up in meetings. For example, people always ask something like, “Have you considered using…” or “Why don’t you try…”. However, they barely give you a concrete reason for those suggestions (for example, what are the pros and cons of the tools they suggest, comparing to the original one I used). </p>
<p>Yes, the only way to know if a program is suitable for my data is to test it by myself and tweak the parameters if necessary. However, there are numerous programs, so just asking others to use some tools without any specific reason or guideline is not very persuasive and meaningful during the discussion. <del>Of course, if you nod off during the meeting but unfortunately still have to say something, anyway, this is a way for doing it.</del></p>
<p>Therefore, one of my goals recently is to have perceptions of those frequently-asked questions, about how to chose tools especially. In this post, I will record some thoughts about the decision making of <strong>RNA-seq aligners</strong>.</p>
<p>First, let’s see how many aligners are available now. You can find an astonishing summary figure (i.e. a super long one!!!!) right in the middle of <a href="https://www.ecseq.com/support/ngs/what-is-the-best-ngs-alignment-software" target="_blank" rel="noopener">this article</a>. This article is talking about the topic which is very relevant to my post: “What is the best NGS alignment software”. Since it was written in 2016, I would say there should be some updates after then (i.e. more aligners! Ones have an interest in the exact number can check the gigantic list on Wiki).</p>
<p>Nonetheless, don’t worry! This list can be further narrow down to the length that somewhat I can handle. Let’s consider some criteria:</p>
<ol>
<li>It should be <strong>splice aware</strong>, for I would like to align the RNA-seq reads to the <strong>genome</strong>.</li>
<li>It is still used broadly and maintained by the authors.</li>
<li>I prefer the tools with lighter computational load and higher speed.</li>
</ol>
<p>Then, I checked the latest articles in regard to the comparison of the RNA-seq aligners to see some convincing and up-to-date evidence. Surprisingly, there are only a few articles talking about this question, and the latest two are:</p>
<blockquote>
<p>2016, Baruzzo <em>et al.</em> <a href="https://www.nature.com/articles/nmeth.4106" target="_blank" rel="noopener">“Simulation-based comprehensive benchmarking of RNA-seq aligners”</a></p>
</blockquote>
<blockquote>
<p>2019, Raplee <em>et al.</em> <a href="https://www.mdpi.com/2075-4426/9/2/18" target="_blank" rel="noopener">“Aligning the Aligners: Comparison of RNA Sequencing Data Alignment and Gene Expression Quantification Tools for Clinical Breast Cancer Research”</a></p>
</blockquote>
<p>I recommend using Baruzzo’s article as the guideline for choosing aligners and tweaking the parameters, for it demonstrates the results of comprehensive investigations with good experimental design. Moreover, <font color=#800000>the authors even kindly provide the instruction about the parameter tweaking in the supplementary information!</font> On the other hand, although Raplee’s article was published later, it has some defects and only compares HISAT2 and STAR. I will discuss this article in the other post.</p>
<a id="more"></a>
<h2 id="Baruzzo’s-article-2016"><a href="#Baruzzo’s-article-2016" class="headerlink" title="Baruzzo’s article, 2016"></a>Baruzzo’s article, 2016</h2><p>Let’s take a closer look into Baruzzo’s article first. Here are some details of the experimental design:</p>
<ul>
<li>Comparing 14 splice-aware algorithms</li>
<li>Analyzing performance at 3 levels:<ul>
<li>base level</li>
<li>read level: most relevant for <font color=#800000>gene-level quantification</font></li>
<li>junction level: important in reconstructing <font color=#800000>alternative splicing events</font></li>
</ul>
</li>
<li>Default vs. Optimized parameter settings</li>
<li>Simulating data: <ul>
<li>(1) human and (2) malaria parasite <em>Plasmodium falciparum</em> </li>
<li>three levels (T1-T3)<table>
<thead>
<tr>
<th>Levels</th>
<th>Substitution</th>
<th>Indel</th>
<th>Error rates</th>
<th>Example</th>
</tr>
</thead>
<tbody><tr>
<td>T1</td>
<td>0.001</td>
<td>0.0001</td>
<td>0.005</td>
<td>Human</td>
</tr>
<tr>
<td>T2</td>
<td>0.005</td>
<td>0.002</td>
<td>0.01</td>
<td>Model organisms</td>
</tr>
<tr>
<td>T3</td>
<td>0.03</td>
<td>0.005</td>
<td>0.02</td>
<td></td>
</tr>
</tbody></table>
</li>
</ul>
</li>
<li>Other analyses, e.g. multimappers, adapters, and two-pass mode </li>
</ul>
<p>And yes, there is a one-line suggestion (I guess that’s what most readers search first for when reading this kind of article) in the discussion section: </p>
<blockquote>
<p>“Based on this analysis the most reliable general-purpose aligners appear to be CLC, Novoalign, GSNAP, and STAR.”</p>
</blockquote>
<p><strong>However</strong>, don’t jump into the conclusion! Be aware, this sentence is in the <strong>discussion</strong> section rather in the abstract or introduction…</p>
<p>One presumption for this one-line summary is you would <font color=#800000>only use the default settings without any tweaking</font>. According to Figure 3 in the article, some aligners can perform pretty well after adjusting some parameters. Therefore, you may want to consider the speed of the aligning for making the decision (Figure 4 in the article); for example, HISAT2 perform far faster than others.</p>
<blockquote class="imgur-embed-pub" lang="en" data-id="UVQegWJ"><a href="//imgur.com/UVQegWJ" target="_blank" rel="noopener"></a></blockquote><script async src="//s.imgur.com/min/embed.js" charset="utf-8"></script>

<p><font color=#808080><em>Figure 3 in Baruzzo’s article. Note that parameter tweaking is necessary for some of the aligners (e.g. HISAT2). For others, teaking is optional but can improve the results (e.g. STAR).</em></font></p>
<p>Here are several points I feel important and take into account after reading:</p>
<ol>
<li><p>In fact, all aligners perform well for the T1 libries with default, expect for CRAC. </p>
<ul>
<li>That means you can use whatever you like for the high quality sequencing of humans.</li>
<li>Even for T2 (certain model organisms with good annotations), most aligners with default setting can handle the task well.</li>
</ul>
</li>
<li><p>Some aligners are remarkably accurate even for the very short anchors and without annotation, and therefore outperform others at junction level. Thus, if the isoform discovery is what you want to do, consider those aligners: HISAT, HISAT2, and ContexMap2.</p>
</li>
<li><p>Some aligners are GUI (graphical user interface), and some of them require licence to use.</p>
</li>
<li><p>With annotations, 2-pass model is not required (save the time!). </p>
</li>
<li><p>Some famous aligners (e.g. TopHat2) are not good choice!</p>
</li>
</ol>
<p><font color=#800000><strong>Eventually, I chose STAR and HISAT2.</strong></font> Here are some notes for making this decision:</p>
<ul>
<li><strong>STAR</strong>: Pretty fast and balance for most situations. Tweaking can further improve the performance.</li>
<li><strong>HISAT2</strong>: Super fast, but the tweaking is necessary (I have done this before so it’s OK for me).</li>
</ul>
<ul>
<li>CLC: GUI; I don’t think my laptop can run it very fast.</li>
<li>Novoalign: Licence required.</li>
<li>GSNAP: Our computing cluster (FARM) does not have the latest version. The version we have is version 2017-11-15. Maybe I can try it in the future.</li>
</ul>
<p>Besides, I also considered BBMap as some posts recommanded it on forums and it is still updating. I wonder why it is not included in Baruzzo’s article because I know JGI is still using BBMap. So maybe I will try it someday.  </p>
<h2 id="Raplee’s-article-2019"><a href="#Raplee’s-article-2019" class="headerlink" title="Raplee’s article, 2019"></a>Raplee’s article, 2019</h2><p>This article only compared STAR and HISAT2 with default setting, and I will discuss it in the other post. The most important discovery in this article is it showed that the <font color=#800000>default HISAT2 cannot handle pseudogenes very well</font>, and this issue became very servere under some circanstances. This finding also imply the importance of using different tools to analyze data.</p>
<hr>
<h2 id="Suggestions-on-Forums"><a href="#Suggestions-on-Forums" class="headerlink" title="Suggestions on Forums"></a>Suggestions on Forums</h2><p>Reading the discussion on forums (SEQanswers and Biostars) provided me some ideas as lots of users share their experience and results. However, I found there are some pitfalls and should be aware of some points:</p>
<ol>
<li>The <strong>date</strong> of the post.<ul>
<li>Bear in mind the fast-evolving nature of NGS!</li>
</ul>
</li>
<li>Check more posts of the users who responded the questions you have interest in.<ul>
<li>To know if the guy is an expert or just a passerby.</li>
</ul>
</li>
<li>The data set and the organism discussed in the post. <ul>
<li>The strategy could be very different for distinct experimental materials.</li>
</ul>
</li>
<li>Evidence talks.<ul>
<li>Go through the example in the post carefully. Some of them may not be very reasonable…</li>
</ul>
</li>
</ol>
<hr>
<h2 id="Othe-articles"><a href="#Othe-articles" class="headerlink" title="Othe articles"></a>Othe articles</h2><p>Just like I mentioned above, I am surprised when I found that there are not so many articles related to the aligner comparison. For whom it may be interesting, here are some older articles, and it is worth mentioning that none of them include HISAT/HISAT2, two of the fastest aligners curently: </p>
<blockquote>
<p>2014, Wang <em>et al.</em><a href="https://ieeexplore.ieee.org/document/7045249" target="_blank" rel="noopener">“Comparisons and performance evaluations of RNA-seq alignment tools”</a>: Tophat, STAR, MapSplice, and GNSAP</p>
</blockquote>
<blockquote>
<p>2014, Benjamin <em>et al.</em><a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-570" target="_blank" rel="noopener">“Comparing reference-based RNA-Seq mapping methods for non-human primate data”</a>: (splice-aware aligners) TopHat2 and GSNAP</p>
</blockquote>
<blockquote>
<p>2013, Engström <em>et al.</em><a href="https://www.nature.com/articles/nmeth.2722" target="_blank" rel="noopener">“Systematic evaluation of spliced alignment programs for RNA-seq data”</a>: (splice-aware aligners) GSNAP, MapSplice, PALMapper, ReadsMap, STAR and TopHat</p>
</blockquote>
<blockquote>
<p>2012, Fonseca <em>et al.</em><a href="https://academic.oup.com/bioinformatics/article/28/24/3169/245777" target="_blank" rel="noopener">“Tools for mapping high-throughput sequencing data”</a>: a long list</p>
</blockquote>
<p>Also, you can check the rollout papers of aligners because the authors have to persuade editors that their new tools have some advantage over others. Besides, some articles did not compare the aligners directly, but went through the entire quantative procedure to do the comparison.</p>

      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/Dry-lab/" rel="tag"># Dry_lab</a>
          
            <a href="/tags/RNA-seq/" rel="tag"># RNA-seq</a>
          
            <a href="/tags/Aligner/" rel="tag"># Aligner</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2020/06/19/20200619-Factors-affecting-the-alignment-of-RNA-seq-reads/" rel="next" title="Factors affecting the alignment of RNA-seq reads">
                <i class="fa fa-chevron-left"></i> Factors affecting the alignment of RNA-seq reads
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
              <a href="/2020/10/10/20201010-FARM-Singularity/" rel="prev" title="Using Singularity on UCD FARM Cluster">
                Using Singularity on UCD FARM Cluster <i class="fa fa-chevron-right"></i>
              </a>
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          


          

  
    <div class="comments" id="comments">
      <div id="disqus_thread">
        <noscript>
          Please enable JavaScript to view the
          <a href="https://disqus.com/?ref_noscript" target="_blank" rel="noopener">comments powered by Disqus.</a>
        </noscript>
      </div>
    </div>

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            Table of Contents
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            Overview
          </li>
        </ul>
      

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image"
                src="https://i.imgur.com/gJGAo55.jpg"
                alt="Po-Kai" />
            
              <p class="site-author-name" itemprop="name">Po-Kai</p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          <nav class="site-state motion-element">

            
              <div class="site-state-item site-state-posts">
              
                <a href="/archives/%7C%7C%20archive">
              
                  <span class="site-state-item-count">6</span>
                  <span class="site-state-item-name">posts</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-categories">
                <a href="/categories/index.html">
                  <span class="site-state-item-count">4</span>
                  <span class="site-state-item-name">categories</span>
                </a>
              </div>
            

            
              
              
              <div class="site-state-item site-state-tags">
                <a href="/tags/index.html">
                  <span class="site-state-item-count">5</span>
                  <span class="site-state-item-name">tags</span>
                </a>
              </div>
            

          </nav>

          

          
            <div class="links-of-author motion-element">
                
                  <span class="links-of-author-item">
                    <a href="https://github.com/loubya" target="_blank" title="GitHub">
                      
                        <i class="fa fa-fw fa-github"></i>GitHub</a>
                  </span>
                
            </div>
          

          
          

          
          

          

        </div>
      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#Baruzzo’s-article-2016"><span class="nav-number">1.</span> <span class="nav-text">Baruzzo’s article, 2016</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Raplee’s-article-2019"><span class="nav-number">2.</span> <span class="nav-text">Raplee’s article, 2019</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Suggestions-on-Forums"><span class="nav-number">3.</span> <span class="nav-text">Suggestions on Forums</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Othe-articles"><span class="nav-number">4.</span> <span class="nav-text">Othe articles</span></a></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      
        <div class="back-to-top">
          <i class="fa fa-arrow-up"></i>
          
            <span id="scrollpercent"><span>0</span>%</span>
          
        </div>
      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Po-Kai</span>

  
</div>


  <div class="powered-by">Powered by <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a></div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">Theme &mdash; <a class="theme-link" target="_blank" href="https://github.com/iissnan/hexo-theme-next">NexT.Pisces</a> v5.1.4</div>




        







        
      </div>
    </footer>

    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  












  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>
  

  
  
    <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.4"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.4"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.4"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.4"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.4"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.4"></script>



  


  

    
      <script id="dsq-count-scr" src="https://pokainote.disqus.com/count.js" async></script>
    

    
      <script type="text/javascript">
        var disqus_config = function () {
          this.page.url = 'https://loubya.github.io/2020/07/02/20200702-Finding-suitable-aligners-for-my-project/';
          this.page.identifier = '2020/07/02/20200702-Finding-suitable-aligners-for-my-project/';
          this.page.title = 'Finding suitable aligners for my project';
        };
        var d = document, s = d.createElement('script');
        s.src = 'https://pokainote.disqus.com/embed.js';
        s.setAttribute('data-timestamp', '' + +new Date());
        (d.head || d.body).appendChild(s);
      </script><!-- hexo-inject:begin --><!-- hexo-inject:end -->
    

  




	





  














  





  

  

  

  
  

  
  


  

  

</body>
</html>
